tests/trace/dsl/test_query.py

```diff
@@ -1,302 +1,768 @@
 
-Context = namedtuple("Context", "span_id trace_id")
-Span = namedtuple("Span", "context parent_id attributes")
+def test_select_all(session: Session) -> None:
+    # i.e. `get_spans_dataframe`
+    sq = SpanQuery()
+    expected = pd.DataFrame(
+        {
+            "context.span_id": ["234", "345", "456", "567"],
+            "context.trace_id": ["012", "012", "012", "012"],
+            "parent_id": ["123", "234", "234", "234"],
+            "name": ["root span", "embedding span", "retriever span", "llm span"],
+            "span_kind": ["UNKNOWN", "EMBEDDING", "RETRIEVER", "LLM"],
+            "status_code": ["OK", "OK", "OK", "ERROR"],
+            "status_message": ["okay", "no problemo", "okay", "uh-oh"],
+            "start_time": [
+                datetime.fromisoformat("2021-01-01T00:00:00.000+00:00"),
+                datetime.fromisoformat("2021-01-01T00:00:00.000+00:00"),
+                datetime.fromisoformat("2021-01-01T00:00:05.000+00:00"),
+                datetime.fromisoformat("2021-01-01T00:00:20.000+00:00"),
+            ],
+            "end_time": [
+                datetime.fromisoformat("2021-01-01T00:00:30.000+00:00"),
+                datetime.fromisoformat("2021-01-01T00:00:05.000+00:00"),
+                datetime.fromisoformat("2021-01-01T00:00:20.000+00:00"),
+                datetime.fromisoformat("2021-01-01T00:00:30.000+00:00"),
+            ],
+            "attributes.input.value": ["210", None, "xyz", None],
+            "attributes.output.value": ["321", None, None, None],
+            "attributes.llm.token_count.prompt": [None, None, None, 100.0],
+            "attributes.llm.token_count.completion": [None, None, None, 200.0],
+            "attributes.metadata": [None, {"a.b.c": 123, "1.2.3": "abc"}, None, None],
+            "attributes.embedding.model_name": [None, "xyz", None, None],
+            "attributes.embedding.embeddings": [
+                None,
+                [
+                    {"embedding.vector": [1, 2, 3], "embedding.text": "123"},
+                    {"embedding.vector": [2, 3, 4], "embedding.text": "234"},
+                ],
+                None,
+                None,
+            ],
+            "attributes.retrieval.documents": [
+                None,
+                None,
+                [
+                    {"document.content": "A", "document.score": 1.0},
+                    {"document.content": "B", "document.score": 2.0},
+                    {"document.content": "C", "document.score": 3.0},
+                ],
+                None,
+            ],
+            "events": [[], [], [], []],
+        }
+    ).set_index("context.span_id", drop=False)
+    actual = sq(session, project_name="abc")
+    assert_frame_equal(
+        actual.sort_index().sort_index(axis=1),
+        expected.sort_index().sort_index(axis=1),
+    )
+    del sq, actual, expected
 
-SPAN_ID = "span_id"
-TRACE_ID = "trace_id"
 
+def test_select(session: Session) -> None:
```

Comment:
Non-blocking suggestion: It would be nice to split these into separate these cases, since it looks like there are three distinct tests here. It's possible to group them under a class like `TestSelect` with method names like `test_select_token_count_prompt`. Easier to step through, and has the added benefit of providing clearer and more granular feedback when a test fails.

Reply:
good idea. i didn't realize you can do that